### PR TITLE
Faster import part 3

### PR DESCRIPTION
**Pull request by danielhollas**
Some minor wins by lazy import of `pgsu`, `yaml` and `urllib.requests` modules.
NOTE: Some of the wins here for `verdi` are dependent on getting rid of `jsonschema` package.

### `import aiida`

```python
Benchmark 1: PYTHONPATH=/home/bl22441/atmospec/aiida-core python -c 'import aiida'
  Time (mean ± σ):      95.2 ms ±   7.9 ms    [User: 82.7 ms, System: 12.5 ms]
  Range (min … max):    75.2 ms … 101.1 ms    30 runs
 
  Warning: Statistical outliers were detected. Consider re-running this benchmark on a quiet system without any interferences from other programs. It might help to use the '--warmup' or '--prepare' options.
 
Benchmark 2: python -c 'import aiida'
  Time (mean ± σ):     118.6 ms ±   1.7 ms    [User: 102.3 ms, System: 15.9 ms]
  Range (min … max):   115.4 ms … 121.3 ms    24 runs
 
Summary
  'PYTHONPATH=/home/bl22441/atmospec/aiida-core python -c 'import aiida'' ran
    1.25 ± 0.10 times faster than 'python -c 'import aiida''
```

### `import aiida.orm`

```python
Benchmark 1: PYTHONPATH=/home/bl22441/atmospec/aiida-core python -c 'import aiida.orm'
  Time (mean ± σ):     261.1 ms ±   1.6 ms    [User: 959.8 ms, System: 2568.4 ms]
  Range (min … max):   258.1 ms … 263.0 ms    11 runs
 
Benchmark 2: python -c 'import aiida.orm'
  Time (mean ± σ):     316.3 ms ±  32.7 ms    [User: 1023.6 ms, System: 2853.6 ms]
  Range (min … max):   283.0 ms … 369.5 ms    10 runs
Summary
  'PYTHONPATH=/home/bl22441/atmospec/aiida-core python -c 'import aiida.orm'' ran
    1.21 ± 0.13 times faster than 'python -c 'import aiida.orm''
```

### `import aiida.cmdline.commands`

```python
Benchmark 1: PYTHONPATH=/home/bl22441/atmospec/aiida-core python -c 'import aiida.cmdline.commands'
  Time (mean ± σ):     240.1 ms ±   2.4 ms    [User: 219.1 ms, System: 20.5 ms]
  Range (min … max):   238.0 ms … 245.7 ms    12 runs
 
Benchmark 2: python -c 'import aiida.cmdline.commands'
  Time (mean ± σ):     258.9 ms ±   3.2 ms    [User: 233.1 ms, System: 25.3 ms]
  Range (min … max):   253.7 ms … 262.7 ms    11 runs
Summary
  'PYTHONPATH=/home/bl22441/atmospec/aiida-core python -c 'import aiida.cmdline.commands'' ran
    1.08 ± 0.02 times faster than 'python -c 'import aiida.cmdline.commands''
```

### CODE REVIEW

**Comment by danielhollas** [comment]
@sphuber This is ready for review. I have no idea why the docs build is failing though...

File: aiida/manage/external/postgres.py [hunk 1]
```diff
@@ -23,8 +23,6 @@
 if TYPE_CHECKING:
     from aiida.manage.configuration import Profile
 
-__all__ = ('Postgres', 'PostgresConnectionMode', 'DEFAULT_DBINFO')
```
Comment (danielhollas) [inline aiida/manage/external/postgres.py:26]: Again, this is in principle breaking API change, but these are only used in a couple of places, chiefly in `verdi quicksetup` command.

**Comment by sphuber** [comment]
> I have no idea why the docs build is failing though...

```
aiida/engine/daemon/execmanager.py:docstring of aiida.engine.daemon.execmanager.upload_calculation:: WARNING: py:class reference target not found: MappingType
```
You can just add `py:class MappinType` to the `docs/source/nitpick-exceptions`

**Comment by danielhollas** [comment]
Thanks @sphuber. That indeed fixed it, although I still don't quite understand why this new warning appeared in this PR specifically, since I was not touching the `MappingType`.

**Comment by sphuber** [comment]
> Thanks @sphuber. That indeed fixed it, although I still don't quite understand why this new warning appeared in this PR specifically, since I was not touching the `MappingType`.

These docstring reference errors are notoriously vague and hard to debug. Which is why we have the ignore list, otherwise we would be stuck forever since they crop up so often for unexplicable reason.

I think the reason for the error is because we use an alias `MappingType` in the doc string of `upload_calculation` instead of the actual import `typing.Mapping`. I think this causes Sphinx to fail to resolve it. Why this happens only on this PR, I am not sure. Maybe because it uses a cache and had a successful build for the module from a previous build, but since now the file is touched, it rebuilt it causing the error to show. Anyway, it is not important, so let's just ignore the warning.

**Review by sphuber — APPROVED** [review]
Thanks @danielhollas

**Pull request merged**